src/control/server/storage/bdev/backend.go

```diff
@@ -225,43 +230,68 @@ func (sb *spdkBackend) formatNvme(req FormatRequest) (*FormatResponse, error) {
 		return nil, errors.New("empty results from spdk binding format request")
 	}
 
+	if err := sb.writeNvmeConfig(req); err != nil {
+		return nil, errors.Wrap(err, "write spdk nvme config")
+	}
+
 	return sb.formatRespFromResults(results)
 }
 
-// Format initializes the SPDK environment, defers the call to finalize the same
-// environment and calls private format() routine to format all devices in the
-// request device list in a manner specific to the supplied bdev class.
-//
-// Remove any stale SPDK lockfiles after format.
-func (sb *spdkBackend) Format(req FormatRequest) (resp *FormatResponse, err error) {
-	defer func() {
-		if err != nil {
-			return
+func (sb *spdkBackend) formatAioFile(req *FormatRequest) (*FormatResponse, error) {
+	resp := &FormatResponse{
+		DeviceResponses: make(DeviceFormatResponses),
+	}
+
+	for _, path := range req.DeviceList {
+		devResp := new(DeviceFormatResponse)
+		resp.DeviceResponses[path] = devResp
+		if err := createEmptyFile(sb.log, path, req.DeviceFileSize); err != nil {
+			devResp.Error = FaultFormatError(path, err)
+			continue
 		}
-		if errConf := sb.writeNvmeConfig(&req); errConf != nil {
-			err = errors.Wrap(errConf, "write spdk nvme config")
+		if err := os.Chown(path, req.OwnerUID, req.OwnerGID); err != nil {
+			devResp.Error = FaultFormatError(path, errors.Wrapf(err,
+				"failed to set ownership of %q to %d.%d", path,
+				req.OwnerUID, req.OwnerGID))
 		}
-	}()
+	}
 
-	// TODO (DAOS-3844): Kick off device formats parallel?
-	switch req.Class {
-	case storage.BdevClassKdev, storage.BdevClassFile, storage.BdevClassMalloc:
-		resp = &FormatResponse{
-			DeviceResponses: make(DeviceFormatResponses),
-		}
+	if err := sb.writeNvmeConfig(req); err != nil {
+		return nil, errors.Wrap(err, "write spdk nvme config")
+	}
 
-		for _, device := range req.DeviceList {
-			resp.DeviceResponses[device] = new(DeviceFormatResponse)
-			sb.log.Debugf("%s format for non-NVMe bdev skipped on %s", req.Class, device)
-		}
+	return resp, nil
+}
 
-		return resp, nil
-	case storage.BdevClassNvme:
-		if len(req.DeviceList) == 0 {
-			return nil, errors.New("empty pci address list in nvme format request")
-		}
+// TODO: implement kdev fs format
```

Comment:
What does this TODO mean? What kind of formatting needs to be done for kdev?

Reply:
added ticket number to provide more detail